syncstorage/tests/functional/test_storage.py

```diff
@@ -1286,6 +1287,49 @@ def test_pagination_with_newer_and_sort_by_oldest(self):
                 self.assertEquals(sorted(int(item['id']) for item in items),
                                   range(start + 1, NUM_ITEMS))
 
+    def test_transactions(self):
+        endpoint = self.root + '/storage/col2'
+        bso1 = {'id': '12', 'payload': 'elegance'}
+        bso2 = {'id': '13', 'payload': 'slovenly'}
+        bsos = [bso1, bso2]
+        self.app.post_json(endpoint, bsos)
+
+        bso3 = {'id': 'a', 'payload': 'internal'}
+        bso4 = {'id': 'b', 'payload': 'pancreas'}
+        resp = self.app.post_json(endpoint + '?batch=true', [bso3, bso4])
+        batch = resp.json["batch"]
+        # This is sooooo going to break when it's run at just the wrong time
+        # and will result in a seriously confused person at the keyboard.
+        guess = float(resp.headers["X-Weave-Timestamp"])
+        self.assertEqual(batch / 1000, math.floor(guess))
+
+        # make sure we don't have the pending commit
+        resp = self.app.get(endpoint)
+        res = resp.json
+        res.sort()
+        self.assertEquals(res, ['12', '13'])
+        self.assertEquals(int(resp.headers['X-Weave-Records']), 2)
+
+        bso5 = {'id': 'c', 'payload': 'tinsel'}
+        bso6 = {'id': '13', 'payload': 'portnoy'}
+        commit = '?batch={0}&commit=true'.format(batch)
+        resp = self.app.post_json(endpoint + commit, [bso5, bso6])
+        self.assertEquals(resp.json["modified"],
+                          float(resp.headers['X-Weave-Timestamp']))
+
+        # make sure the changes apply
+        resp = self.app.get(endpoint)
+        res = resp.json
+        res.sort()
+        self.assertEquals(res, ['12', '13', 'a', 'b', 'c'])
+        self.assertEquals(int(resp.headers['X-Weave-Records']), 5)
+        resp = self.app.get(endpoint + '/13')
+        self.assertEquals(resp.json["payload"], 'portnoy')
```

Comment:
In fact, @rnewman, I encourage you to think of any specific edge-case-style interactions that might affect this API, and we can add specific tests for them.